### DISCUSSION TITLE
CrateDB as Full text Search engine for Documents, and emails.

### DISCUSSION THREAD

**Discussion by sreekanthedayar** [crate/crate#17110] [Q&A] [answered]
Hi As discussed, we are a small startup building a worksplace collaboration platform that primarily deals with emails, messages, documents and web content. Our primary DB is My SQL and We store all documents in Cloudflare R2. Currently we are evaluating a dedicated search engine that can scale horizontally with commodity hardware. We narrow down our search to Manticore and Crate DB, infact crate db is much better as we can also use this for analytics. 

Will cratedb fits well with this use case? I was able to spin up a 3 node cluster on hetzner without much effort, this make me so impressed with how easy to manage the deployment unlike other popular alternate. 

Do you have any benchmarks against any dedicated search engine like opensearch or elastic (as crate is based on elastic), or manticore. Also how it will be compared against inmemory datastore search engines like Meallisearch or typesense? eventhough our requirement is not ultra low latency search, just curious on how it performs against popular search engines. 

As we want to go with onpremise, we will be considering opensource version and will subscribe to any support plans as we grow .

**Comment by hlcianfagna** [comment]
Hi,

> Do you have any benchmarks against any dedicated search engine like opensearch

The first thing that comes to mind is a run of `rockbench` a colleague did a few months ago where he got these results showing lower latencies across the board:

```
Benchmarking Elastic
Generator identifier:  thyb6zEs7J
Initial sleep of 4s and polling period of 25s
Sleep done. Now issuing requests to calculate e2e latency.
Latency: 1.019278s
Latency: 1.026499s
Latency: 1.036756s
Latency: 1.026115s
Latency: 1.023782s
Latency: 1.040088s
Latency: 1.02546s
Latency: 1.026143s
Latency: 1.029186s
^C
signal received: interrupt
2024/07/03 19:42:07 done
Benchmarking CrateDB
Generator identifier:  YkpJ0rknGL
Initial sleep of 2s and polling period of 25s
Sleep done. Now issuing requests to calculate e2e latency.
Latency: 1.050514s
Latency: 106.732ms
Latency: 88.5ms
Latency: 90.912ms
Latency: 95.79ms
Latency: 135.561ms
Latency: 88.375ms
Latency: 100.388ms
Latency: 90.523ms
```

**Reply by sreekanthedayar** [reply]
Thank you for quick response.

**Comment by geragray** [comment, marked as answer]
@sreekanthedayar thanks a lot for the interest. In addition to data that @hlcianfagna already shared we have a few additional resources with recent benchmarks
https://cratedb.com/blog/how-cratedb-compares-to-rockbench
https://cratedb.com/blog/independent-time-series-benchmark-confirms-cratedb-top-tier-performance

May I ask if self-managed on-prem is a hard requirement for you? Even though it is quite easy to run CrateDB at scale by yourself, a fully managed CrateDB Cloud is usually a more cost-effective solution for small and medium-sized companies where workloads grow over time and you don't have a dedicated SRE team (example https://cratedb.com/stories/govspend). Would be happy to understand more what requirements you have and how we can help.

**Reply by sreekanthedayar** [reply]
In the course of product development, some of our government departments shown keen interest to deploy this on premises, So we thought of starting on premise model and battle test the product before going fully SaaS model. As these departments are strict on data ownership and always prefer onpremise deployment model, I picked the open source. We are from India, and here in India, majority of government like to use opensource.

But I have already impressed with your S3 import module in cloud. So definitely we will keep cloud solution also in our mind, first I have to convince our development team to start working on this. Right now Iam doing indepth analysis on all the pros and cons, so that i can convince my team on adopting this. Once we integrate opensource, its matter of business decision to go with cloud for our SaaS product offerings. 
Meanwhile i'd also saw the jepsen test, which stated the dataloss, hope such issues had been rectified in recent versions.

**Reply by geragray** [reply]
Thanks a lot for the context. If you are required to use some sort of Gov Cloud that is not relying on AWS/Azure/GCP underneath, then indeed self-managing the entire deployment is the best way to start. Core functionality is the same, but in the cloud, we have a better focus on ease of use (fully managed backups, upgrades, security, etc) and a handful of data ingestion tools from NoSQL databases which are usually not suited for analytical workloads.

Jepsen folks evaluated CrateDB in 2016 which resulted in https://aphyr.com/posts/332-jepsen-crate-0-54-9-version-divergence. Since then we created missing transparency docs https://cratedb.com/docs/crate/reference/en/latest/appendices/resiliency.html, spent a significant amount of effort ensuring consistency of distributed data through testing done on our CI, and hired Jepsen to verify it. We have a handful of very large companies relying on CrateDB for years (https://cratedb.com/stories) including a few household names from India (but they are not public yet). Having said that we have been considering running an external evaluation one more time during the first half of 2025.

**Reply by sreekanthedayar** [reply]
Thanks for the time to elaborate this in detail. Yes, my first thing is to make our team alligned with this product, once that is done, we will start working on this,  I will keep you posted may be after little time. I'm planning to start integration possibly from janaury mid onwards. As it's completely new, i have to convince them. 
Once again thank you for the detailed info.
